main.go

```diff
@@ -65,6 +75,21 @@ func main() {
 
 	server := &http.Server{}
 
+	// Setup signal handling for graceful shutdown
+	sigs := make(chan os.Signal, 1)
```

Comment:
I get the motivation, but the default behaviour for SIGINT/SIGTERM is to terminate the process. There's no need to clean up the collector.